### PR TITLE
atlantis: init at 0.10.1

### PR DESCRIPTION
**Pull request by jpotier**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/cluster/atlantis/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
```
Comment (jonringer) [inline pkgs/applications/networking/cluster/atlantis/default.nix]: looks like the repo has a go.mod, did you want try making this into a buildGoModule? Should remove need for deps.nix

Comment (jpotier) [reply pkgs/applications/networking/cluster/atlantis/default.nix]: That is my very first packaging of a go package. Actually my very first interaction with go at all, if one can consider this an interaction. I wasn't even aware this was possible! I'll change that immediately, better not commit thousands of useless lines of text.

Comment (jonringer) [reply pkgs/applications/networking/cluster/atlantis/default.nix]: https://nixos.org/nixpkgs/manual/#ssec-go-modules

Comment (jpotier) [reply pkgs/applications/networking/cluster/atlantis/default.nix]: Fixed it, and actually read the nixpkgs manual section about it, I'm too used to not rely on documentation, so I missed this. Should be better now!

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executable shows usage and version correctly
repo looks active and well maintained

```
[4 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73278
1 package were build:
atlantis
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build atlantis

File: pkgs/applications/networking/cluster/atlantis/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/applications/networking/cluster/atlantis/default.nix]: that's the point of reviews :) don't worry about it

Comment (jonringer) [reply pkgs/applications/networking/cluster/atlantis/default.nix]: and it's only applicable to projects that converted over modules

**Pull request merged**